### PR TITLE
feat: add runAsNonRoot:true to default

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>

## Description

I noticed that we have a check for runAsNonRoot in validation rules but we don't have any mutation for that.

This PR will fix that.

kindly ping @chipzoller

Does it make sense? 

## Checklist

- [x] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [x] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.

### CODE REVIEW

File: other/add-default-securitycontext/add-default-securitycontext.yaml [hunk 1]
```diff
@@ -22,6 +22,7 @@ spec:
       patchStrategicMerge:
         spec:
           securityContext:
+            +(runAsNonRoot): true
```
Comment (chipzoller) [inline other/add-default-securitycontext/add-default-securitycontext.yaml:25]: Have you tested this that it works properly without setting to `"true"`? The spirit of this is good, just need to make sure it works as written (boolean) and doesn't require string.

**Review by chipzoller — COMMENTED** [review]
Before approving, need to test and validate on at least the current 1.5.x release and preferably also 1.6-dev-latest.

File: other/add-default-securitycontext/add-default-securitycontext.yaml (hunk 1 above)
Comment (developer-guy) [reply other/add-default-securitycontext/add-default-securitycontext.yaml]: I've updated it as "true" 🙋🏻‍♂️

Comment (chipzoller) [reply other/add-default-securitycontext/add-default-securitycontext.yaml]: So does it work?

**Comment by developer-guy** [comment]
> Before approving, need to test and validate on at least the current 1.5.x release and preferably also 1.6-dev-latest.

I've tested it on locally and it worked as expected 👇 

It should be a boolean btw, I got an error when I change it as string
> kyverno-568449c9d6-fdptz kyverno E0131 16:19:10.359522       1 mutation.go:86] WebhookServer "msg"="mutate error" "error"="failed to validate resource mutated by policy add-default-securitycontext: ValidationError(io.k8s.api.core.v
1.Pod.spec.securityContext.runAsNonRoot): invalid type for io.k8s.api.core.v1.PodSecurityContext.runAsNonRoot: got \"string\", expected \"boolean\"" "action"="mutate" "gvk"="/v1, Kind=Pod" "operation"="CREATE" "resource"="default/Pod/"


Screen Shot 2022-01-31 at 19 14 44
Screen Shot 2022-01-31 at 19 22 31

File: other/add-default-securitycontext/add-default-securitycontext.yaml (hunk 1 above)
Comment (developer-guy) [reply other/add-default-securitycontext/add-default-securitycontext.yaml:25]: It should be a boolean btw, I got an error when I change it as string

> kyverno-568449c9d6-fdptz kyverno E0131 16:19:10.359522 1 mutation.go:86] WebhookServer "msg"="mutate error" "error"="failed to validate resource mutated by policy add-default-securitycontext: ValidationError(io.k8s.api.core.v
1.Pod.spec.securityContext.runAsNonRoot): invalid type for io.k8s.api.core.v1.PodSecurityContext.runAsNonRoot: got "string", expected "boolean"" "action"="mutate" "gvk"="/v1, Kind=Pod" "operation"="CREATE" "resource"="default/Pod/"

Comment (chipzoller) [reply other/add-default-securitycontext/add-default-securitycontext.yaml:25]: Ok

**Review by chipzoller — APPROVED** [review]
Thanks for the addition here!

**Pull request merged**